src/main/java/at/hannibal2/skyhanni/features/misc/skillprogress/SkillProgress.kt

```diff
@@ -0,0 +1,138 @@
+package at.hannibal2.skyhanni.features.misc.skillprogress
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.api.SkillAPI
+import at.hannibal2.skyhanni.api.SkillAPI.activeSkill
+import at.hannibal2.skyhanni.api.SkillAPI.lastUpdate
+import at.hannibal2.skyhanni.api.SkillAPI.showDisplay
+import at.hannibal2.skyhanni.api.SkillAPI.skillMap
+import at.hannibal2.skyhanni.api.SkillAPI.stackMap
+import at.hannibal2.skyhanni.events.ConfigLoadEvent
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.events.PreProfileSwitchEvent
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.onToggle
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.NumberUtil.formatNumber
+import at.hannibal2.skyhanni.utils.NumberUtil.roundToPrecision
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStringsAndItems
+import at.hannibal2.skyhanni.utils.SpecialColour
+import at.hannibal2.skyhanni.utils.StringUtils.firstLetterUppercase
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import io.github.moulberry.notenoughupdates.util.Utils
+import net.minecraft.init.Items
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+import kotlin.math.ceil
+import kotlin.time.Duration.Companion.seconds
+
+class SkillProgress {
+
+    private val config get() = SkyHanniMod.feature.misc.skillProgressDisplayConfig
+    private var skillExpPercentage = 0.0
+    private var display = emptyList<List<Any>>()
+    private val defaultStack = Utils.createItemStack(Items.banner, "Default")
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (!showDisplay) return
+        config.position.renderStringsAndItems(display, itemScale = 1.5, posLabel = "Skill Progress")
```

Comment:
could you change it to use renderRenderables. Not needed, but I want switch away from those random render functions that all do the same thing.

Reply:
Sent you a PM on discord